server/src/main/java/org/elasticsearch/cluster/coordination/Coordinator.java

```diff
@@ -1874,7 +1873,7 @@ public void onFailure(Exception e) {
                     ackListener.onNodeAck(getLocalNode(), exception); // other nodes have acked, but not the master.
                     publishListener.onFailure(exception);
                 }
-            }, EsExecutors.DIRECT_EXECUTOR_SERVICE, transportService.getThreadPool().getThreadContext());
```

Comment:
There's no need to capture the thread context here, everything we do here is already automatically in system context.